### ISSUE TITLE
Can simple elastix register for bmp images?

### ISSUE THREAD

**Issue by JerrySongCST** [SuperElastix/SimpleElastix#292]
Hi, everyone. I'm brand new to simpleelastix, I see many pics' shown in the examples is in 'nii' format. However, what I want to do is to register two 2d x-ray images in the format of 'bmp'. I need to draw the bounding box in the moving image, can someone give me some tips? Really appreciate.
The following is what I have tried, I referenced "Error after run elastix.Execute() #10" on how to register 2d images,  but seems not work:
import SimpleITK as sitk

fixed_image = sitk.ReadImage("001.bmp", sitk.sitkFloat32)
fixedImage2d = sitk.Extract(fixed_image, (fixed_image.GetWidth(), fixed_image.GetHeight(), 0), (0, 0, 0))

moving_image = sitk.ReadImage("001.bmp", sitk.sitkFloat32)
moving_image2d = sitk.Extract(moving_image, (moving_image.GetWidth(), moving_image.GetHeight(), 0), (0, 0, 0))

elastixImageFilter = sitk.ElastixImageFilter()
elastixImageFilter.SetFixedImage(fixedImage2d)
elastixImageFilter.SetMovingImage(moving_image2d)
elastixImageFilter.SetParameterMap(sitk.GetDefaultParameterMap("rigid"))
elastixImageFilter.Execute()

result_image = elastixImageFilter.GetResultImage()
transformParameterMap = elastixImageFilter.GetTransformParameterMap()

result_image = sitk.Cast(result_image, sitk.sitkUInt8)
sitk.WriteImage(result_image, 'result.bmp')
elastixImageFilter.PrintParameterMap()

**Comment by kaspermarstal** [comment]
Yes, you can register bmp. Once your image is read with ReadImage, SimpleElastix does not care in what format the image is stored on disk. To register 2d images, try to set "FixedImagePyramid" and "MovingImagePyramid" to  "RecursiveImagePyramid".

**Comment by reventropy** [comment]
When you are building the code it guesses at what python version you're using.  This means that if you use a different version than what the system uses natively you might run into this error.  The way I did this was to create a virtual environment in my favourite editor (Pycharm).  I then activated the environment in terminal and did the build/make/install steps from there so that SimpleElastix would guess to use the python virtual environment I'm actively using.  The issue is really in the instructions here: https://simpleelastix.readthedocs.io/GettingStarted.html#compiling-on-linux.  If you want to install on a virtual environment you need to do all the steps from there rather than just the last one (as suggested).
